src/sentry/api/endpoints/sentry_app_components.py

```diff
@@ -43,13 +44,15 @@ def get(self, request, organization):
                 _components = _components.filter(type=request.GET['filter'])
 
             for component in _components:
-                sentry_app_components.Preparer.run(
-                    component=component,
-                    install=install,
-                    project=project,
-                )
-
-            components.extend(_components)
+                try:
+                    sentry_app_components.Preparer.run(
+                        component=component,
+                        install=install,
+                        project=project,
+                    )
+                    components.append(component)
+                except APIError:
+                    continue
```

Comment:
Would a metric be useful here? Could help in putting partners on the naughty list.

Reply:
Yea, absolutely. Good call.

Reply:
Actually, we would have logs for when the downstream object fails. So we could derive this information from that. I don't know if we need to explicitly do something here.

Reply:
Sounds good. :+1: